### PR TITLE
#284: throw an error on server error response

### PR DESCRIPTION
**Pull request by dr0bz**
Penthouse isn't stopping nor is throwing an error when the server is responding with an error status code. With this change only 200-299 response status codes will let penthouse proceed.

### CODE REVIEW

**Comment by pocketjoso** [comment]
Allright, looks good overall. However I thought about this a bit more, and to avoid introducing a breaking change _and_ to support all existing use cases (imagine someone wanting to create critical css for their 404 page), I suggest we refactor this a bit to introduce this through a new option. If you're up for it I'll let you decide how to model it, my criteria are just:
- the default value should work as today; i.e. doesn't throw on _any_ status code
- easy to configure to only accept `2xx` status codes, or `2xx` + `404`

OK for you?

File: test/basic.test.js [hunk 1]
```diff
@@ -4,17 +4,39 @@ import path from 'path'
...
 
 import normaliseCss from './util/normaliseCss'
 
+const serverPort = 8888;
+
 function staticServerFileUrl (file) {
   return 'file://' + path.join(process.env.PWD, 'test', 'static-server', file)
 }
 
+function responseSatusUrl (code) {
```
Comment (pocketjoso) [inline test/basic.test.js]: small typo here in function name.

Comment (dr0bz) [reply test/basic.test.js]: Yes, i'll fix that.

**Comment by dr0bz** [comment]
Yes, I fully agree with you: it should be optional. Disallowing 3xx responses could also break things. 

> or `2xx` + `404`

Do you mean here it could be a mix of codes?

I would suggest an option name `acceptedResponseCode` with following value types `string|number`. 
- String value: `success`
  - means all `2xx` codes
- Number value: concrete http response status code for ex. `200`

What do you think?

**Comment by pocketjoso** [comment]
I think a rehex would be more powerful, it would be easy to handle all
cases. Alternatively it could also be a callback function where either the
status code or the full response is provided, and the return value should
be true or false to signify whether the response is ok to proceed with or
not. Wdyt?

Le dim. 10 févr. 2019 14:00, Hermann Bier <notifications@github.com> a
écrit :

**Comment by pocketjoso** [comment]
friendly ping, if you want to have this merged please check my last comment. :)

**Comment by dr0bz** [comment]
Hi @pocketjoso,

sorry for the delay. Your suggestion regarding regex/function is fine. What about to allow all of the option values:

`allowedResponseCode`: `number|regex|function` ?

Regards,
dr0bz

**Comment by pocketjoso** [comment]
Wonder if the number is really useful, as with a single number you would only be able to support exactly the `200` status, not any ranges or combinations of statuses. In the end I'll leave it up to you, but I think `regex|function` might be enough.

**Comment by dr0bz** [comment]
I've implemented the number option as well. I think the number option is handy, if you expect a particular status code.

Best regards,
dr0bz

**Pull request merged**